### PR TITLE
Create value classes for Content Packs

### PR DESCRIPTION
**Pull request by kmerz**
## Description
Replace the big content pack hash with a value class.

## Motivation and Context
Value classes help to ensure default values, encapsulate accessors and help find bugs while writing code.

## How Has This Been Tested?
Clicked (hopefully) through all scenarios. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: graylog2-web-interface/src/components/content-packs/ContentPackSelection.test.jsx [hunk 1]
```diff
@@ -43,25 +44,17 @@ describe('<ContentPackSelection />', () => {
...
-        name: 'name',
-        summary: 'summary',
-        description: 'descr',
-        vendor: 'vendor',
-        url: 'url',
-      };
       called += 1;
-      if (called === Object.keys(contentPack).length) {
-        expect(state.contentPack).toEqual(contentPack);
+      if (called === 5) {
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/content-packs/ContentPackSelection.test.jsx]: What happens when one of the change callbacks does not fire? The following `expect()` calls will never be executed then.

File: graylog2-web-interface/src/logic/content-packs/ContentPack.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+      vendor,
+      url,
+      requires,
+      parameters,
+      entities,
+    }));
+  }
+
+  toObject() {
+    return this._value;
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/logic/content-packs/ContentPack.js]: I think we should never expose the internal state of the object directly, it makes the whole object mutable. We should either clone the state before returning it, or (way better in this case) implement a `toJSON` function that returns a new object containing the desired serialization format for JSON (As AFAIK `toObject` is only used in conjunction with `JSON.stringify`).

File: graylog2-web-interface/src/components/content-packs/ContentPackEdit.test.jsx [hunk 3]
```diff
@@ -46,19 +38,16 @@ describe('<ContentPackEdit />', () => {
   const selectedEntities = { dashboard: [{ id: '111-beef', type: 'dashbaord' }] };
 
   const parameter = { title: 'Port', name: 'PORT', type: 'integer', default_value: '23' };
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/content-packs/ContentPackEdit.test.jsx]: Maybe it is beyond the scope of this PR, but maybe we should introduce a `Parameter` value class as well?

Comment (kmerz) [reply graylog2-web-interface/src/components/content-packs/ContentPackEdit.test.jsx]: I did a value class for Entity in a following PR already. And will continue to transfer at need. (Which will come sooner or later!)

File: graylog2-web-interface/src/components/content-packs/ContentPackSelection.test.jsx (hunk 1 above)
Comment (kmerz) [reply graylog2-web-interface/src/components/content-packs/ContentPackSelection.test.jsx]: So this expect will fail:
`expect(changeFn.mock.calls.length).toBe(5);`

File: graylog2-web-interface/src/components/content-packs/ContentPackPreview.jsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ class ContentPackPreview extends React.Component {
   };
 
   _renderDownload() {
-    const contentPackEncoded = encodeURIComponent(JSON.stringify(this.props.contentPack, null, 2));
+    const contentPackEncoded = encodeURIComponent(JSON.stringify(this.props.contentPack.toJSON(), null, 2));
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/content-packs/ContentPackPreview.jsx]: You do not need to call `toJSON` explicitly, `JSON.stringify` is doing it [implicitly](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Description).

File: graylog2-web-interface/src/components/content-packs/ContentPackSelection.test.jsx (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/components/content-packs/ContentPackSelection.test.jsx]: Why don't you check the state of `state.contentPack` immediately after that expect? To me it looks like there is an unnecessary level of asynchronicity.

File: graylog2-web-interface/src/components/content-packs/ContentPackPreview.jsx (hunk 4 above)
Comment (kmerz) [reply graylog2-web-interface/src/components/content-packs/ContentPackPreview.jsx]: I did not know that! Thank you!

File: graylog2-web-interface/src/components/content-packs/ContentPackVersions.test.jsx [hunk 5]
```diff
@@ -4,39 +4,39 @@ import { mount } from 'enzyme';
...
+    .summary('This is a summary')
+    .vendor('graylog.com')
+    .url('www.graylog.com');
+
   const contentPack = {
-    1: buildPack(1),
-    2: buildPack(2),
-    3: buildPack(3),
-    4: buildPack(4),
+    1: contentPackRev.rev(1).build().toJSON(),
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/components/content-packs/ContentPackVersions.test.jsx]: I think we don't need `toJSON` here either, right?

Comment (kmerz) [reply graylog2-web-interface/src/components/content-packs/ContentPackVersions.test.jsx]: You are absolutely right!!

**Pull request merged**